Source/WebCore/platform/mac/DataDetectorHighlight.mm

```diff
@@ -213,7 +229,13 @@ bool areEquivalent(const DataDetectorHighlight* a, const DataDetectorHighlight*
     if (!a || !b)
         return false;
 
-    return a->type() == b->type() && a->range() == b->range();
+    bool result = a->type() == b->type();
```

Comment:
Seems we could simplify this function a bit:
```
if (a->type() != b->type())
    return false;
return !a->isRangeSupportingType() || a->range() == b->range();
```

Reply:
True, I went with my version for readability but I think they're equally readable 🤷 Fixing

Reply:
Mine is slightly more efficient :P